scraper/target1337x.go

```diff
@@ -4,62 +4,96 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/PuerkitoBio/goquery"
 	"github.com/qascade/yast/movie"
 
 	colly "github.com/gocolly/colly"
 )
 
+// scrape1337x returns the scraped results in Result type
 func (s *Scraper) scrape1337x(context *QueryContext) (results []Result, err error) {
 
+	var links []string
+	var movielist []movie.Movie
 	results = make([]Result, 0)
+
 	url := fmt.Sprintf("https://1337x.to/search/%s/1/", context.Query)
+
 	s.collector.OnHTML("tr", func(e *colly.HTMLElement) {
-		movieMetaDataDOM := e.DOM
-		anchorTagDom := movieMetaDataDOM.Find("a").Siblings()
 
-		var magnet string
-		if magnetPageUrl, exists := anchorTagDom.Attr("href"); exists {
-			magnet = getMagnetLink("https://1337x.to" + magnetPageUrl)
+		doc := e.DOM
+		anchortags := doc.Find("a").Siblings()
+
+		if attr, exists := anchortags.Attr("href"); exists {
+			if strings.Contains(attr, "/torrent/") && !strings.HasPrefix(attr, "http://") {
+				link := "https://1337x.to" + attr
+				links = append(links, link)
+			}
 		}
+		newmovie := movieFromString(doc)
+		movielist = append(movielist, newmovie)
+
+	})
+	s.collector.OnHTML("div.col-9 div.box-info", func(h *colly.HTMLElement) {
 
-		newMovie := movieFromString(movieMetaDataDOM.Text(), magnet)
-		newMovie.Magnet = magnet
-		results = append(results, newMovie)
+		metadata := h.DOM
 
+		movieName := metadata.Find("div.box-info-heading h1").Text()
+		metadata.Find("a").Each(func(i int, s *goquery.Selection) {
+			if magnet, exists := s.Attr("href"); exists {
+				if strings.Contains(magnet, "magnet") {
+					for i, movie := range movielist {
+						if compareNamesfromdifferentPages(strings.TrimSpace(movie.Name), strings.TrimSpace(movieName)) {
```

Comment:
rename  `compareNamesfromdifferentPages` -> `compareNamesFromDiffPages`. We need to maintain proper Camel Casing.

Reply:
Also, can you explain why do we need this function?

Reply:
movie struct is populated from "tr" (table rows) , i compare the name of the movie from page 1 to page 2 in order to insert the magnet later when found , but the problem i faced was on page 2 ,1337x limits the name of the movie till 65 characters and append `...` to it. Hence created a function to compare movie name from different webpages